src/ApparentHorizons/ComputeExcisionBoundaryVolumeQuantities.tpp

```diff
@@ -0,0 +1,206 @@
+                          metric);
+  gr::shift(make_not_null(&shift), psi, inv_metric);
+  gr::lapse(make_not_null(&lapse), shift, psi);
+}
+
+/// Dual frame case
+template <typename SrcTagList, typename DestTagList, typename TargetFrame>
+void ComputeExcisionBoundaryVolumeQuantities::apply(
+    const gsl::not_null<Variables<DestTagList>*> target_vars,
+    const Variables<SrcTagList>& src_vars, const Mesh<3>& /*mesh*/,
+    const Jacobian<DataVector, 3, TargetFrame, Frame::Inertial>& jacobian,
+    const InverseJacobian<DataVector, 3, Frame::ElementLogical, TargetFrame>&
+        /*inverse_jacobian*/) {
+
+  static_assert(
+      std::is_same_v<tmpl::list_difference<SrcTagList, allowed_src_tags>,
+                     tmpl::list<>>,
+      "Found a src tag that is not allowed");
+  static_assert(
+      std::is_same_v<tmpl::list_difference<required_src_tags, SrcTagList>,
+                     tmpl::list<>>,
+      "A required src tag is missing");
+
+  static_assert(
+      std::is_same_v<
+          tmpl::list_difference<DestTagList, allowed_dest_tags<TargetFrame>>,
+          tmpl::list<>>,
+      "Found a dest tag that is not allowed");
+  static_assert(
+      std::is_same_v<
+          tmpl::list_difference<required_dest_tags<TargetFrame>, DestTagList>,
+          tmpl::list<>>,
+      "A required dest tag is missing");
+
+  const auto& psi =
+      get<gr::Tags::SpacetimeMetric<3, Frame::Inertial>>(src_vars);
+
+  if (target_vars->number_of_grid_points() !=
+      src_vars.number_of_grid_points()) {
+    target_vars->initialize(src_vars.number_of_grid_points());
+  }
+
+  using metric_tag = gr::Tags::SpatialMetric<3, TargetFrame>;
+  using inv_metric_tag = gr::Tags::InverseSpatialMetric<3, TargetFrame>;
+  using lapse_tag = gr::Tags::Lapse<DataVector>;
+  using inertial_shift_tag = gr::Tags::Shift<3, Frame::Inertial>;
+
+  // Additional temporary tags used for multiple frames
+  using inertial_metric_tag = gr::Tags::SpatialMetric<3, Frame::Inertial>;
+  using inertial_inv_metric_tag =
+      gr::Tags::InverseSpatialMetric<3, Frame::Inertial>;
+
+  // All of the temporary tags, including some that may be repeated
+  // in the target_variables (for now).
+  using full_temp_tags_list =
+      tmpl::list<metric_tag, inv_metric_tag, lapse_tag, inertial_shift_tag,
+                 inertial_metric_tag,
+                 inertial_inv_metric_tag>;
+
+  // temp tags without variables that are already in DestTagList.
+  using temp_tags_list =
+      tmpl::list_difference<full_temp_tags_list, DestTagList>;
+  TempBuffer<temp_tags_list> buffer(get<0, 0>(psi).size());
+
+  // These may or may not be temporaries, depending on if they are asked for
+  // in target_vars.
+  auto& lapse = *(get<lapse_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_shift = *(get<inertial_shift_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_metric = *(get<inertial_metric_tag>(
```

Comment:
Probably not your fault because you took what was in the other file, but can you name the `metric` and `inv_metric` `spatial_metric` and `inv_spatial_metric`? Similarly, `psi` -> `spacetime_metric`. Just do a find-replace in this file in the order `inv_metric` -> `inv_spatial_metric`, `inertial_metric` -> `inertial_spatial_metric`, `psi` -> `spacetime_metric`. Doing just `metric` will need to be manual because find-replace will probably hit sub-words